src/ape/api/projects.py

```diff
@@ -113,10 +114,55 @@ def _create_manifest(
         if version:
             manifest.version = version
 
+        manifest.compilers = cls._create_compilers_list(source_paths, contract_types)
         manifest.sources = cls._create_source_dict(source_paths, contracts_path)
         manifest.contract_types = contract_types
+
         return manifest
 
+    @classmethod
+    def _create_compilers_list(
+        cls,
+        source_paths: List[Path],
+        contract_types: Dict[str, ContractType],
+    ):
+        compiler_list: List[Compiler] = []
+        contracts_folder = cls.config_manager.contracts_folder
+        for ext, compiler in cls.compiler_manager.registered_compilers.items():
+            sources = [x for x in source_paths if x.suffix == ext]
+            if not sources:
+                continue
+
+            try:
+                version_map = compiler.get_version_map(sources, contracts_folder)
+            except APINotImplementedError:
+                versions = list(compiler.get_versions(sources))
```

Comment:
I just thought of a way we can handle this:

```python
for src in sources:
    versions = compiler.get_versions([src])
    if not versions:
        continue
    
    version = max(versions)
```

Reply:
loop through the sources and this way we can handle situations where there happen to be more than one version albeit likely less accurate than an implemented `get_version_map()` API method

Reply:
realized this is not possible due to issues such as when the source does not list a pragma and the version is determined another way.

the only way to handle multi-version is to implement `get_version_map()`